### PR TITLE
adds mass mindswap, a new wizard spell

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Adds a new spell for wizards, "mass mindswap". Transfers the mind of all living mobs except for the caster within a 2 screen radius of the wizard. 3 minute cooldown. 
2 Wizpoints cost 

## Why It's Good For The Game
Wizard loves chaos, but kind of falls flat due to the fact it majorly relies on hardstuns and cheese tactics. Denth was talking about this yesterday and I think it's a really good idea, causes some confusion but people can sort themselves out. A really fun part of this spell is that it allows all living mobs to transfer, so you might get turned into a pAI or something. 

Some kinks need to be ironed out, but this should be fun and funny 

## Testing
Compiles with minor testing currently
## Changelog
:cl: GDN, Mcramon
add: Added a new wizard spell, "Mass mindswap", sprite made by mcramon
/:cl:

### CODE REVIEW

**Comment by datlo** [comment]
Reading the spell code, I cant tell if this only targets mobs with a player or any mobs at all. You probably dont want to have players mindswapped with playerless pets or whatever.

**Comment by GDNgit** [comment]
> Reading the spell code, I cant tell if this only targets mobs with a player or any mobs at all. You probably dont want to have players mindswapped with playerless pets or whatever.

Requires a player and a non ssd one at that

**Comment by SpringSkipper** [comment]
> A really fun part of this spell is that it allows all living mobs to transfer, so you might get turned into a pAI or something.

If a wizard gets turned into a simplemob then the round ends. You need to address that before including non-carbon mobs in the raffle.

Also, you need to keep in mind that a mindswap is effectively perfect cover. There is no way to beat this without metagaming that someone isn't who they say they are. Even if you are damned sure that someone is the real wizard (because some other guy calls them out as not being who they say) it becomes a space law dead zone where there's no clear ruling because there's no tangible proof other than what they say. Keep in mind also that the wizard can cast spells while stunned, stamcritted, paralyzed, or anything less than fully unconscious. In practice, this means the wizard will cast this, try to blend in, and either be utterly undetectable because of the nature of mindswaps and just wait 5 minutes before rampaging again or else the wizard will be outed and attempt to get by the next 5 minutes anyways by hiding behind technicalities of spacelaw until they can re-shuffle everyone. I think a lot of people will also be hesitant to call others out because it feels like bwoink bait.

**Comment by GDNgit** [comment]
> If a wizard gets turned into a simplemob then the round ends. You need to address that before including non-carbon mobs in the raffle.

As mentioned in the PR, the caster (or wizard) is unaffected by this spell

**Comment by SpringSkipper** [comment]
> As mentioned in the PR, the caster (or wizard) is unaffected by this spell

Ohhh, I overlooked that.
Okay yeah... so total opposite then. 1 SP and a 3 minute cooldown tops. It doesn't really debilitate or impair people in any major way (unless swapping them with someone who was already debilitated or impaired). Before the spell it's everyone against the wizard and after the spell it's still everyone against the wizard. Since it doesn't swap protected roles I don't believe this even lets you swap apprentices into crew members. Part of the reason for the hardstuns and cheese tactics over funni chaos stuff is because of unreasonable pricing like this. The only reason I suggest 3 minutes and not 30 seconds is the huge AoE.

This might require a clearer ruling on wizard minion weirdness for what someone's allegiance should be if they get mindswapped into (or out of) being a soulstone or necrostone minion.

**Comment by TravisAngeI** [comment]
maybe the spellcaster could be effected, but make a check that makes sure they're placed in a carbon body or just not affected?

dunno much about code so no idea how hard it might be

File: code/datums/spells/mind_transfer.dm [hunk 1]
```diff
@@ -71,11 +72,45 @@ Also, you never added distance checking after target is selected. I've went ahea
...
+	if(length(caster.mind.special_verbs))//If they had any special verbs, we add them here.
 		for(var/V in caster.mind.special_verbs)
 			caster.verbs += V
-	//MIND TRANSFER END
 
-	//Here we paralyze both mobs and knock them out for a time.
-	caster.Paralyse(paralysis_amount_caster)
-	victim.Paralyse(paralysis_amount_victim)
+/obj/effect/proc_holder/spell/aoe/mass_mindswap //Basically Mass Ranged Mindswap
+	name = "Mass mindswap"
```
Comment (Henri215) [inline code/datums/spells/mind_transfer.dm:80]:
```suggestion
	name = "Mass Mindswap"
```
Like it is with other spells.

**Comment by SteelSlayer** [comment]
This is being TM'd because the teams would like to see it in action before they decisively vote.

**Comment by SteelSlayer** [comment]
I will be closing this PR. It has 2 objections, and no in favor, and has been open for a very long time. 

### Objection reasons: ###

**Team member 1**: This seems like a bit of a nightmare Administratively. Thinking on it more, this is just gonna make people cryo when they get knocked out of their character. I already dislike mindswap, makin' it mass sounds :gatto:

**Team member 2**: I think something similiar like this would be interesting, but not exactly in this format. Perhaps a one time use spell that would mindswap everyone on Z1? Basically like the one time use of mass summon guns/magic.

**Team member 3**: This has the potential to be either really bad or really good, I'm leaning towards the former. Would like to see this TM'd

**Team member 4**: Same as Matt

**Pull request closed without merging**